R/net.mod.init.R

```diff
@@ -136,26 +136,39 @@ initialize.net <- function(x, param, init, control, s) {
 #'
 init_status.net <- function(dat) {
 
+  type <- get_control(dat, "type", override.null.error = TRUE)
+  nsteps <- get_control(dat, "nsteps")
+  tgl <- get_control(dat, "tergmLite")
+  vital <- get_param(dat, "vital")
+  groups <- get_param(dat, "groups")
+  if (type %in% c("SIS", "SIR")){
+    rec.rate <- get_param(dat, "rec.rate")
+  }
+  if (vital) {
+    di.rate <- get_param(dat, "di.rate")
+  }
+
   # Variables ---------------------------------------------------------------
-  i.num <- dat$init$i.num
-  i.num.g2 <- dat$init$i.num.g2
-  r.num <- dat$init$r.num
-  r.num.g2 <- dat$init$r.num.g2
+  i.num <- get_init(dat, "i.num", override.null.error = TRUE)
+  if (type  == "SIR"){
+    r.num <- get_init(dat, "r.num", override.null.error = TRUE)
+  }
 
-  status.vector <- dat$init$status.vector
-  num <- sum(dat$attr$active == 1)
-  #TODO: check that this works for tergmLite
-  statOnNw <- "status" %in% dat$temp$nwterms
+  status.vector <- get_init(dat, "status.vector", override.null.error = TRUE)
+  num <- sum(get_attr(dat, "active") == 1)
 
-  groups <- dat$param$groups
   if (groups == 2) {
-    group <- dat$attr$group
+    group <- get_attr(dat, "group")
+    i.num.g2 <- get_init(dat, "i.num.g2")
+    if (type  == "SIR"){
+      r.num.g2 <- get_init(dat, "r.num.g2")
+    }
   } else {
     group <- rep(1, num)
   }
 
-  type <- dat$control$type
-
+  #TODO: check that this works for tergmLite
+  statOnNw <- "status" %in% dat$temp$nwterms
```

Comment:
You can remove that TODO line now; that is resolved

Reply:
Removed.